### ISSUE TITLE
Encoding slices is broken on 1.22 when using structs pointer fields - AllWithContext

### ISSUE THREAD

**Issue by amit-evermile** [guregu/dynamo#230]
Hi, we are using a struct with pointer fields, example:
```
type DynamoObj struct {
	Id                   string                `dynamo:"id,hash"`
	OrderId              *string               `dynamo:"order_id"`
	CommunicationType    *string               `dynamo:"communication_type"`
	CreatedAt            int64                 `dynamo:"created_at,unixtime"`
}
```

When using 
```
result := make([]DynamoObj, 0)
table.Get("id", id).AllWithContext(ctx, &result)
```

The pointer fields (OrderId, CommunicationType) are overridden with the values from the last record read.
I've verified that this issue doesn't happen on 1.21.

**Comment by chriselkins** [comment]
I'm also having this issue. It only started with 1.22 and previous versions did not have this issue.

**Comment by guregu** [comment]
Sorry about this. I have just released v1.22.1 which should fix this. Please try it out and let me know how it goes.

**Comment by nightlord189** [comment]
Thank you, @guregu. It was fixed for me.
